pkg/utils/time_test.go

```diff
@@ -0,0 +1,79 @@
+package utils
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestParseDuration(t *testing.T) {
+	tests := []struct {
+		name    string
+		s       string
+		want    time.Duration
+		wantErr bool
+	}{
+		{
+			name:    "duration string's time unit is 'd'",
+			s:       "1d",
+			want:    time.Hour * 24 * time.Duration(1),
+			wantErr: false,
+		},
+		{
+			name:    "duration string's time unit is not 'd'",
+			s:       "300ms",
+			want:    300 * time.Millisecond,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ParseDuration(tt.s)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseDuration() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("ParseDuration() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestParseTimestamp(t *testing.T) {
+	tim := time.Unix(1, 0)
+	tests := []struct {
+		name    string
+		ts      string
+		want    time.Time
+		wantErr bool
+	}{
+		{
+			name:    "parsed incorrect",
+			ts:      "",
+			want:    time.Now(),
+			wantErr: true,
+		},
+		{
+			name:    "parsed correct",
+			ts:      "1",
+			want:    tim,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ParseTimestamp(tt.ts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseTimestamp() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.name == "parsed incorrect" {
```

Comment:
Please use another way to implement this.

`tt.name == "parsed incorrect"`

Reply:
I've tried some other ways, but the `if != nil`'s situation will return a dynamic real date time, which makes the `want` value hard to be defined. Is there any other way to make a test case?

Reply:
done